osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -371,8 +382,57 @@ protected override void Update()
                 new Dimension(),
                 new Dimension(GridSizeMode.Relative, 0.5f, minSize: 500, maxSize: 700 + widescreenBonusWidth * 300),
             };
+
+            updateDebounce();
+        }
+
+        #region Selection debounce
+
+        private BeatmapInfo? debounceQueuedSelection;
+        private double debounceElapsedTime;
+
+        private void debounceQueueSelection(BeatmapInfo beatmap)
+        {
+            debounceQueuedSelection = beatmap;
+            debounceElapsedTime = 0;
+        }
+
+        private void updateDebounce()
+        {
+            if (debounceQueuedSelection == null) return;
+
+            // avoid debounce running early if there's a single long frame.
+            debounceElapsedTime += Math.Min(1000 / 60.0, Clock.ElapsedFrameTime);
```

Comment:
Not sure how I feel about this. Maybe it works for a single long frame, but if someone is, say, only able to pull 30fps in song select this will effectively double the debounce duration. Thinking about especially about mobile performance here.

Reply:
valid concern, i'll adjust.